### PR TITLE
Added fixes mentioned in previous PR

### PR DESCRIPTION
**Pull request by z3t0**
This is still Arduino 1.0 but is labelled as latest for now (will be renamed later)

### CODE REVIEW

**Comment by sudar** [comment]
I see that for library keywords it is still using `syn keyword` and is not grouping them to use a different group. Did you fix it?

On second thought I think it's not a big deal. If you are not working on it, then I think I can go merge this, after making some changes to the credits and readme files.

**Comment by z3t0** [comment]
Oh sorry it was a misunderstanding I can easily change that, I thought you
wanted to change the part after keyword,  like arduinoConstant to
arduinoLibraryConstant.
I'll do that soon and make another PR

On Thursday, February 26, 2015, Sudar Muthu <notifications@github.com>
wrote:
> I see that for library keywords it is still using syn keyword and is not
grouping them to use a different group. Did you fix it?
>
> On second thought I think it's not a big deal. If you are not working on
it, then I think I can go merge this, after making some changes to the
credits and readme files.
>
> —
> Reply to this email directly or view it on GitHub.<
https://ci3.googleusercontent.com/proxy/hTp_snctobpT7tjL49rhuqthxeC-drEKfEKEIZNnuY2d06pPKbaCn1dsn1zt-gJ8MswSWZ1_8XtBgJVFgge52hVXnfjFGJKe7CAsYEtrmTwyZAXkU4b5wu0svfz44gddQDyimTctsm6fsHhFBlXscM7XguaXsw=s0-d-e1-ft#https://github.com/notifications/beacon/AIlUfXsvRx9oynealOcWTWj5Vc6Ri4Wxks5nvshIgaJpZM4DmB9R.gif
>

-- 
Thanks,
Rafi Khan

**Comment by z3t0** [comment]
Will fix library keyword and make a new PR

**Comment by sudar** [comment]
Oops.. sorry for the confusion.

As I said it's fine if you don't want to change it.

**Comment by z3t0** [comment]
No problem, I definitely want to change it I just misunderstood what you
meant. I'll get it done asap

On Thu, Feb 26, 2015 at 9:33 AM, Sudar Muthu <notifications@github.com>
wrote:

> Oops.. sorry for the confusion.
>
> As I said it's fine if you don't want to change it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sudar/vim-arduino-syntax/pull/8#issuecomment-76198495>
> .
>



-- 
Thanks,
Rafi Khan

**Comment by z3t0** [comment]
I am not sure

**Comment by z3t0** [comment]
Please merge this.

**Comment by sudar** [comment]
As I said over Google Hangout, ignore my previous comment.

Give me a day or two, I will just test this once in my setup and will merge it.

Thanks for all the effort you have put in this project.

**Comment by z3t0** [comment]
OK sounds good!

On Tuesday, March 3, 2015, Sudar Muthu <notifications@github.com> wrote:
> As I said over Google Hangout, ignore my previous comment.
>
> Give me a day or two, I will just test this once in my setup and will
merge it.
>
> Thanks for all the effort you have put in this project.
>
> —
> Reply to this email directly or view it on GitHub.<
https://ci5.googleusercontent.com/proxy/CNUgE3qJnryIR0zajD9lUqdTwPMK9wacRBhDUE54zdSqBovequkrwvxiO58_CV7Y6GNUhyXcK-tQ29_VeCR028Kx1AJkop4sNIUT15lomF-FRY5ENj20rpNRPa0j1xIgfXShNNUe1esGEVC1D2oToatDc3ZjWw=s0-d-e1-ft#https://github.com/notifications/beacon/AIlUfZJK1TjGB8Cy1uGWRszZYnpKgR8Nks5nxoL0gaJpZM4DmB9R.gif
>

-- 
Thanks,
Rafi Khan

**Comment by sudar** [comment]
@z3t0 Thanks. I just merged it.

**Comment by z3t0** [comment]
@sudar Happy I could help!

**Comment by Jaidan** [comment]
This PR appears to have been completely borked?  Just used and I'm getting syntax errors starting on line 265 every time I load an .ino file.   I don't know jack about creating syntax files, but it appears the lines should all be at least: `syn foo bar` where foo is a group and bar is at least one token to be defined.  It seems like the grouping got nuked.

File: syntax/arduino.vim [hunk 1]
```diff
@@ -28,33 +24,907 @@ else
...
+syn keyword arduinoConstant GSM
+syn keyword  GSMVoiceCall
+syn keyword arduinoConstant GSM_SMS
+syn keyword arduinoConstant GPRS
+syn keyword  GSMClient
+syn keyword  GSMServer
+syn keyword  GSMModem
+syn keyword  GSMScanner
+syn keyword arduinoConstant GSMPIN
+syn keyword
```
Comment (Jaidan) [inline syntax/arduino.vim]: This is clearly broken =(

**Comment by sudar** [comment]
DAMN!! I am reverting it.

Thanks @Jaidan for reporting it.

@z3t0 Can you take a look at it?

**Pull request closed without merging**